Exiled.Events/Handlers/Internal/Round.cs

```diff
@@ -35,7 +35,6 @@ public static void OnRestartingRound()
         {
             MultiAdminFeatures.CallEvent(MultiAdminFeatures.EventType.ROUND_END);
 
-            API.Features.Map.ClearCache();
```

Comment:
Horever, there's a reason for this, the game calls `GC.Collect()` after executing this line, keeping the cache does nothing (I guess) but I'm sure we don't want to prevent objects from being collected. I think also clearing the cache on the MapGenerated event is better.

Reply:
But clearing the cache too early can lead to nullrefs in plugins aswell, if they try to access something after RoundRestarting but before it's restarted, if the list is empty they could perhaps nullref (IE if someone has a Map.Rooms.FirstOrDefault() without a null check)

Reply:
> I don't think postponing the GC.Collect() call a few sec will make a huge difference.
`GC.Collect()` is manually called from the game when the round restarts, so we can't actually postpone that call, but the GC will collect those objects sooner or later anyway, based on their generation.
I don't think that would actually have a big impact on used memory.

Reply:
Calling Map.Rooms after RoundRestarting seems to be incorrect because there are already no players on the server tbh

Reply:
> I'm still sure we should clear the cache after RoundRestarting & MapGenerated.

So, between RoundRestarting and MapGenerated?

Reply:
Yep

Reply:
No, after another implementation I think it shold be cleared during RoundRestarting, plugins shouldn't access the cache after RoumdRestarting, it makes no sence, the issue itself is so controversial.

Reply:
Well one way or another, plugins were accessing Map.Rooms after the cache had been cleared, and therefore causing the rooms in that list to no longer match with actual rooms on the map after OnMapGenerated.

Moving the clearing of the cache and building those lists to 1 frame after the map is generated, solves any and all potential issues that could arise, without any noticable downsides. (yes, the objects in those lists won't be immediately collected by the GC but it's minor, once they are cleared from the lists, the next time GC collects, it'll purge them, so I don't really see the issue)